src/Elm/Arg.elm

```diff
@@ -0,0 +1,72 @@
+module Elm.Arg exposing
+    ( Arg, unit, var, varWith
+    , tuple, triple
+    , record, field
+    , Expression
+    )
+
+{-|
+
+@docs Arg, unit, var, varWith
+
+@docs tuple, triple
+
+@docs record, field
+
+-}
+
+import Elm.Annotation
+import Internal.Arg
+import Internal.Compiler as Compiler
+
+
+{-| -}
+type alias Expression =
+    Compiler.Expression
+
+
+{-| -}
+type alias Arg val =
+    Internal.Arg.Arg val
+
+
+{-| -}
+var : String -> Arg Expression
+var =
+    Internal.Arg.var
+
+
+{-| -}
+varWith : String -> Elm.Annotation.Annotation -> Arg Expression
+varWith =
+    Internal.Arg.varWith
+
+
+{-| -}
+unit : Arg Expression
+unit =
+    Internal.Arg.unit
+
+
+{-| -}
+tuple : Arg Expression -> Arg Expression -> Arg ( Expression, Expression )
```

Comment:
How about `Arg a -> Arg b -> Arg (a, b)` instead? Would help for nesting

Reply:
Whoops, done!